src/connection.cpp

```diff
@@ -350,120 +395,172 @@ namespace oxen::quic
             for (auto it = mid; it != streams.end(); ++it)
             {
                 auto& stream_ptr = it->second;
-                if (stream_ptr and not stream_ptr->sent_fin)
-                    strs.push_back(stream_ptr.get());
+                if (stream_ptr and not stream_ptr->_sent_fin)
+                    channels.push_back(stream_ptr.get());
             }
```

Comment:
We should append the datagram channel right here rather than later, so that it also ends up randomly positioned in the initial channel list.